src/main/java/com/example/javaOjt/services/AuthorService.java

```diff
@@ -59,10 +62,39 @@ public GetAuthorResponse getAuthorById(Integer id, boolean withBooks) {
   }
 
   /**
-   * Fetches the complete list of authors.
+
+   * Fetches the list of authors and sort if needed.
+   *
+   * @param attribute the attribute to sort by (id, name)
+   * @param order the order to sort by (asc, dsc)
+   * @return GetAuthorsResponse containing the list of authors
    */
-  public GetAuthorsResponse getAuthorsList() { // Should return an empty list on empty DB
-    return new GetAuthorsResponse(authorRepository.findAll());
+  public GetAuthorsResponse getAuthorsList(@Nullable AuthorSortBy attribute, @Nullable Order order) { // Should return an empty list on empty DB
+    String attributeString = (attribute == null) ? "id" : attribute.toString();
+    String orderString = (order == Order.DSC) ? "desc" : "asc";
+    GetAuthorsResponse response = new GetAuthorsResponse(authorRepository.authorsList(attributeString, orderString));
```

Comment:
[nits]
このまま return する方が綺麗ですので！
```suggestion
    return new GetAuthorsResponse(authorRepository.authorsList(attributeString, orderString));
```